### PR TITLE
Solver doc fix

### PR DESCRIPTION
**Pull request by JoshuaFisherC**
Updated and fleshed out numpy solver documentation.

### CODE REVIEW

File: gillespy2/solvers/numpy/CLE_solver.py [hunk 1]
```diff
@@ -142,12 +141,16 @@ def run(self=None, model=None, t=20, number_of_trajectories=1, increment=None, s
...
+        :type timeout: int
+        
+        :param resume: Result of a previously run simulation, to be resumed
+        
+        :param tau_tol: Tolerance level for Tau leaping algorithm.  Larger tolerance values will
+            result in larger tau steps. Default value is 0.03.
+        :type tau_tol: float
 
-        :returns:
+        :returns: a list of each trajectory simulated.
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/CLE_solver.py:153]: solver run functions always return result objects not lists.  Also missing the rtype tag for the return type

File: gillespy2/solvers/numpy/ode_solver.py [hunk 2]
```diff
@@ -88,26 +88,41 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
             integrator_options={}, live_output=None, live_output_options={}, timeout=None, resume=None, **kwargs):
         """
         :param model: gillespy2.model class object
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ode_solver.py:90]: Add `(deprecated)` to the description of the model are to indicate that this arg is deprecated

File: gillespy2/solvers/numpy/ode_solver.py [hunk 3]
```diff
@@ -88,26 +88,41 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
         :param live_output_options: dictionary contains options for live_output. By default {"interval":1}.
             "interval" specifies seconds between displaying.
-            "clear_output" specifies if display should be refreshed with each displa
+            "clear_output" specifies if display should be refreshed with each display
+        :type live_output_options:  str
+        
+        :param timeout: If set, if simulation takes longer than timeout, will exit.
+        :type timeout: int
+        
+        :param resume: Result of a previously run simulation, to be resumed
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ode_solver.py:123]: missing type tag for resume arg

File: gillespy2/solvers/numpy/ode_solver.py [hunk 4]
```diff
@@ -88,26 +88,41 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
+        
+        :returns: a list of each trajectory simulated.
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ode_solver.py:125]: solver run functions always return result objects not lists. Also missing the rtype tag for the return type

File: gillespy2/solvers/numpy/ssa_solver.py [hunk 5]
```diff
@@ -58,18 +58,36 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
         Run the SSA algorithm using a NumPy for storing the data in arrays and generating the timeline.
 
         :param model: The model on which the solver will operate.
+        :type model: GillesPy2.model
+        
         :param t: The end time of the solver.
+        :type t: int
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ssa_solver.py:64]: `t` can be an int or float

File: gillespy2/solvers/numpy/tau_hybrid_solver.py [hunk 6]
```diff
@@ -827,6 +827,11 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
             "interval" specifies seconds between displaying.
             "clear_output" specifies if display should be refreshed with each display
         :type live_output_options:  str
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_hybrid_solver.py:829]: `live_output_options` is a dict not str

File: gillespy2/solvers/numpy/tau_hybrid_solver.py [hunk 7]
```diff
@@ -827,6 +827,11 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
+
+        :param timeout: If set, if simulation takes longer than timeout, will exit.
+        :type timeout: int
+        
+        :returns: a list of each trajectory simulated.
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_hybrid_solver.py:834]: solver run functions always return result objects not lists. Also missing the rtype tag for the return type

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 8]
```diff
@@ -129,12 +129,16 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
 
-        :param timeout:
-        :param resume:
-        :param tau_tol:
-        :param kwargs:
+        :param timeout: If set, if simulation takes longer than timeout, will exit.
+        :type timeout: int
 
-        :returns:
+        :param resume: Result of a previously run simulation, to be resumed
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:135]: missing type tag for resume arg

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 9]
```diff
@@ -129,12 +129,16 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
+
+        :param tau_tol: Tolerance level for Tau leaping algorithm.  Larger tolerance values will
+            result in larger tau steps. Default value is 0.03.
+        :type tau_tol: float
+        
+        :returns: a list of each trajectory simulated.
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:141]: solver run functions always return result objects not lists. Also missing the rtype tag for the return type

File: gillespy2/solvers/cpp/tau_leaping_c_solver.py [hunk 10]
```diff
@@ -24,19 +24,71 @@
...
             resume=None, live_output: str = None, live_output_options: dict = {}, tau_tol=0.03, **kwargs):
 
+        """
+        :param model: The model on which the solver will operate. (Deprecated)
+        :type model: gillespy2.Model
+
+        :param t: end time of simulation
+        :type t: int
+
+        :param number_of_trajectories: Should be 1. This is deterministic and will always have same results
```
Comment (BryanRumsey) [inline gillespy2/solvers/cpp/tau_leaping_c_solver.py:57]: This is incorrect

File: gillespy2/solvers/numpy/CLE_solver.py [hunk 11]
```diff
@@ -30,11 +30,13 @@ class CLESolver(GillesPySolver):
...
     A Chemical Langevin Equation Solver for GillesPy2 models.
 
-    TODO: fix
-    This solver uses an algorithm calculates
+    This solver uses an algorithm that calculates
     multiple reactions in a single step over a given tau step size.  The change in propensities
     over this step are bounded by bounding the relative change in state, yielding greatly improved
     run-time performance with very little trade-off in accuracy.
+
+    :param model: The model on which the solver will operate. (Deprecated)
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/CLE_solver.py:38]: Model is not deprecated in the init function

File: gillespy2/solvers/numpy/CLE_solver.py [hunk 12]
```diff
@@ -30,11 +30,13 @@ class CLESolver(GillesPySolver):
...
+    :type model: GillesPy2.model
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/CLE_solver.py:39]: lowercase 'G' and 'P' for types

File: gillespy2/solvers/numpy/ode_solver.py [hunk 13]
```diff
@@ -28,7 +28,11 @@
 
 class ODESolver(GillesPySolver):
     """
-    This Solver produces the deterministic continuous solution via ODE.
+    This Solver produces the deterministic continuous solution via Ordinary Differential Equations.
+    Uses integrators from scipy.integrate.ode to perform calculations used to produce solutions.
+
+    :param model: The model on which the solver will operate. (Deprecated)
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ode_solver.py:34]: Model is not deprecated in the init function

File: gillespy2/solvers/numpy/ode_solver.py [hunk 14]
```diff
@@ -28,7 +28,11 @@
...
+    :type model: GillesPy2.model
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ode_solver.py:35]: lowercase 'G' and 'P' in types

File: gillespy2/solvers/numpy/ssa_solver.py [hunk 15]
```diff
@@ -27,6 +27,14 @@
 
 
 class NumPySSASolver(GillesPySolver):
+
+    """
+    This solver produces simulations of systems via Stochastic Simulation Algorithm.
+
+    :param model: The model on which the solver will operate. (Deprecated)
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ssa_solver.py:34]: Model is not deprecated in the init function

File: gillespy2/solvers/numpy/ssa_solver.py [hunk 16]
```diff
@@ -27,6 +27,14 @@
...
+    :type model: GillesPy2.model
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/ssa_solver.py:35]: lowercase 'G' and 'P' in types

File: gillespy2/solvers/numpy/tau_hybrid_solver.py [hunk 17]
```diff
@@ -63,6 +63,10 @@ class TauHybridSolver(GillesPySolver):
     This Solver uses a root-finding interpretation of the direct SSA method,
     along with ODE solvers to simulate ODE and Stochastic systems
     interchangeably or simultaneously.
+    Uses integrators from scipy.integrate.ode to perform calculations used to produce solutions.
+
+    :param model: The model on which the solver will operate. (Deprecated)
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_hybrid_solver.py:68]: Model is not deprecated in the init function

File: gillespy2/solvers/numpy/tau_hybrid_solver.py [hunk 18]
```diff
@@ -63,6 +63,10 @@ class TauHybridSolver(GillesPySolver):
...
+    :type model: GillesPy2.model
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_hybrid_solver.py:69]: lowercase 'G' and 'P' in types

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 19]
```diff
@@ -32,6 +32,9 @@ class TauLeapingSolver(GillesPySolver):
     multiple reactions in a single step over a given tau step size.  The change in propensities
     over this step are bounded by bounding the relative change in state, yielding greatly improved
     run-time performance with very little trade-off in accuracy.
+
+    :param model: The model on which the solver will operate. (Deprecated)
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:36]: Model is not deprecated in the init function

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 20]
```diff
@@ -32,6 +32,9 @@ class TauLeapingSolver(GillesPySolver):
...
+    :type model: GillesPy2.model
```
Comment (BryanRumsey) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:37]: lowercase 'G' and 'P' in types

**Review by BryanRumsey — CHANGES_REQUESTED** [review]

**Review by BryanRumsey — APPROVED** [review]

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 21]
```diff
@@ -124,17 +129,23 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
         :type live_output_options: dict
 
-        :param timeout:
-        :param resume:
-        :param tau_tol:
-        :param kwargs:
+        :param timeout: If set, if simulation takes longer than timeout, will exit.
+        :type timeout: int
+
+        :param resume: Result of a previously run simulation, to be resumed
```
Comment (mdip226) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:140]: .

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 22]
```diff
@@ -124,17 +129,23 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
...
+        :type resume: gillespy2.Results
 
-        :returns:
+        :param tau_tol: Tolerance level for Tau leaping algorithm.  Larger tolerance values will
+            result in larger tau steps. Default value is 0.03.
+        :type tau_tol: float
+        
+        :returns: A result object containing the results of the simulation
```
Comment (mdip226) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:147]: .

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 23]
```diff
@@ -124,17 +129,23 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
         :param live_output: The type of output to be displayed by solver. Can be "progress", "text", or "graph".
         :type live_output: str
 
-        :param live_output_options: COntains options for live_output. By default {"interval":1}. "interval"
+        :param live_output_options: Contains options for live_output. By default {"interval":1}. "interval"
```
Comment (mdip226) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:132]: newline tab like the others

File: gillespy2/solvers/numpy/CLE_solver.py [hunk 24]
```diff
@@ -137,17 +141,23 @@ def run(self=None, model=None, t=20, number_of_trajectories=1, increment=None, s
         :param live_output: The type of output to be displayed by solver. Can be "progress", "text", or "graph".
         :type live_output: str
 
-        :param live_output_options: COntains options for live_output. By default {"interval":1}. "interval"
+        :param live_output_options: Contains options for live_output. By default {"interval":1}. "interval"
```
Comment (mdip226) [inline gillespy2/solvers/numpy/CLE_solver.py:144]: newline tab like the others

**Review by mdip226 — COMMENTED** [review]
Just minor changes so that there is a uniform format throughout. Nothing major, looks good! Some of my comments are just me thinking and may not require a change at all.

File: gillespy2/solvers/numpy/ssa_solver.py [hunk 25]
```diff
@@ -57,21 +67,40 @@ def run(self=None, model=None, t=None, number_of_trajectories=1, increment=None,
         """
         Run the SSA algorithm using a NumPy for storing the data in arrays and generating the timeline.
```
Comment (mdip226) [inline gillespy2/solvers/numpy/ssa_solver.py:68]: I think you meant to say something like 'Run the SSA algorithm. Uses a NumPy array to store results and for generating the timeline.' Just missed a word after NumPy.

File: gillespy2/solvers/cpp/tau_leaping_c_solver.py [hunk 26]
```diff
@@ -24,19 +24,70 @@
 from .c_solver import CSolver, SimulationReturnCode
 
 class TauLeapingCSolver(GillesPySolver, CSolver):
+
+    """
+    A Tau Leaping solver for GillesPy2 models.  This solver uses an algorithm calculates
```
Comment (mdip226) [inline gillespy2/solvers/cpp/tau_leaping_c_solver.py:29]: algorithm 'that' calculates

File: gillespy2/solvers/numpy/tau_leaping_solver.py [hunk 27]
```diff
@@ -28,10 +28,13 @@
 
 class TauLeapingSolver(GillesPySolver):
     """
-    A Tau Leaping Solver for GillesPy2 models.  This solver uses an algorithm calculates
+    A Tau Leaping solver for GillesPy2 models.  This solver uses an algorithm calculates
```
Comment (mdip226) [inline gillespy2/solvers/numpy/tau_leaping_solver.py:31]: algorithm 'that'

File: gillespy2/solvers/cpp/ode_c_solver.py [hunk 28]
```diff
@@ -24,6 +24,11 @@
 from .c_solver import CSolver, SimulationReturnCode
 
 class ODECSolver(GillesPySolver, CSolver):
+    """
+    This Solver produces the deterministic continuous solution via Ordinary Differential Equations.
+    Uses integrators from scipy.integrate.ode to perform calculations used to produce solutions.
```
Comment (BryanRumsey) [inline gillespy2/solvers/cpp/ode_c_solver.py:29]: Is this correct? I believe we use Sundials for the C++ ode solvers

**Review by BryanRumsey — CHANGES_REQUESTED** [review]

**Review by briandrawert — APPROVED** [review]

**Pull request merged**